modules/gateway/peers.go

```diff
@@ -499,33 +479,7 @@ func (g *Gateway) managedConnectv120Peer(conn net.Conn, remoteVersion string, re
 		return err
 	}
 
-	g.mu.RLock()
-	port := g.port
-	g.mu.RUnlock()
-	// Send our dialable address to the peer so they can dial us back should we
-	// disconnect.
-	err = connectPortHandshake(conn, port)
-	if err != nil {
-		return err
-	}
-
-	g.mu.Lock()
-	defer g.mu.Unlock()
-	g.addPeer(&peer{
-		Peer: modules.Peer{
-			Inbound:    false,
-			Local:      remoteAddr.IsLocal(),
-			NetAddress: remoteAddr,
-			Version:    remoteVersion,
-		},
-		sess: muxado.Client(conn),
-	})
-	// Add the peer to the node list. We can ignore the error: addNode
-	// validates the address and checks for duplicates, but we don't care
-	// about duplicates and we have already validated the address by
-	// connecting to it.
-	g.addNode(remoteAddr)
-	return g.saveSync()
+	return g.managedConnectv100Peer(conn, remoteVersion, remoteAddr)
```

Comment:
yeah. I like this reorg because it makes it clear that v120 is just v100 plus a new handshake.

However, if we merge the smux stuff in at the same time, v120 will no longer be able to share code with v100, so the DRY breaks. Also, I need to remember to update all the v120 references to v130 (assuming our target for the new gateway code is v1.3.0)